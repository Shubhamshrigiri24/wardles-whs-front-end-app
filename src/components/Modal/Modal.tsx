import React, { FC } from "react";
import {
  Dialog,
  Typography,
  makeStyles,
  Theme,
  DialogContent,
} from "@material-ui/core";

export interface ModalProps {
  title: string;
  text?: string;
  isOpen: boolean;
}

const useStyles = makeStyles((theme: Theme) => ({
  modalPadding: {
    padding: theme.spacing(5),
  },
  textRow: {
    marginTop: theme.spacing(2),
    marginBottom: theme.spacing(2),
  },
  title: {
    marginBottom: theme.spacing(2),
    marginTop: theme.spacing(2),
  },
}));

const Modal: FC<ModalProps> = ({ title, text, isOpen, children }) => {
  const classes = useStyles();

  return (
    <Dialog data-testid={"Modal-component"} open={isOpen} maxWidth={"xs"}>
      <DialogContent className={classes.modalPadding}>
        <Typography
          data-testid={"Modal-title-text"}
          className={classes.title}
          variant={"h5"}
        >
          {title}
        </Typography>
        {text && (
          <Typography
            data-testid={"Modal-body-text"}
            className={classes.textRow}
          >
            {text}
          </Typography>
        )}
        {children}
      </DialogContent>
    </Dialog>
  );
};

export default Modal;
