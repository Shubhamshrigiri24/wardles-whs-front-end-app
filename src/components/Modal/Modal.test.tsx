import React from "react";
import { render } from "@testing-library/react";

import Modal, { ModalProps } from "./Modal";

describe("<Modal />", () => {
  const props = makeProps();
  it("should match snapshot'", () => {
    const { getByTestId } = render(<Modal {...props} />);
    const modal = getByTestId("Modal-component");
    expect(modal).toMatchSnapshot();
  });

  it("should contain a title text component", () => {
    const { getByTestId } = render(<Modal {...props} />);
    expect(getByTestId("Modal-title-text").textContent).toBe("Title text");
  });
  it("should contain a body text component", () => {
    const { getByTestId } = render(<Modal {...props} />);
    expect(getByTestId("Modal-body-text").textContent).toBe("Content of modal");
  });
});

function makeProps(v?: Partial<ModalProps>): ModalProps {
  return {
    title: "Title text",
    text: "Content of modal",
    isOpen: true,
  };
}
