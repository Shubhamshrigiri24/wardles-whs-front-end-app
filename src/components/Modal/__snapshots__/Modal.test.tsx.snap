// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should match snapshot' 1`] = `
<div
  class="MuiDialog-root"
  data-testid="Modal-component"
  role="presentation"
  style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-test="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper"
    role="none presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      class="MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthXs MuiPaper-elevation24 MuiPaper-rounded"
      role="dialog"
    >
      <div
        class="MuiDialogContent-root makeStyles-modalPadding-1"
      >
        <h5
          class="MuiTypography-root makeStyles-title-3 MuiTypography-h5"
          data-testid="Modal-title-text"
        >
          Title text
        </h5>
        <p
          class="MuiTypography-root makeStyles-textRow-2 MuiTypography-body1"
          data-testid="Modal-body-text"
        >
          Content of modal
        </p>
      </div>
    </div>
  </div>
  <div
    data-test="sentinelEnd"
    tabindex="0"
  />
</div>
`;
